security/generate.go

```diff
@@ -21,18 +22,19 @@ type (
 	// ServiceData contains the data necessary to render the secure endpoints
 	// constructor.
 	ServiceData struct {
-		// Name is the name of the service.
-		Name string
+		*httpcodegen.ServiceData
```

Comment:
Why do we need to use transport specific types? this code should work for other transports as well and it won't with these changes it seems?